modules/EvManager/main/CommandRegistry.hpp

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#pragma once
+
+#include <cstddef>
+#include <everest/logging.hpp>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+namespace module::main {
+
+class car_simulatorImpl;
+template <typename Func> class RegisteredCommand;
+
+class RegisteredCommandBase;
+
+class CommandRegistry {
+public:
+    CommandRegistry(car_simulatorImpl* const simulatorIn) : simulator{simulatorIn} {
+    }
+
+    // TODO(MarzellT): maybe we could deduce the argument count from the function signature somehow
+    template <typename FunctionT>
+    void registerCommand(std::string commandName, size_t argumentCount, FunctionT&& function) {
+        registeredCommands.try_emplace(
+            commandName, std::make_unique<RegisteredCommand<FunctionT>>(simulator, commandName, argumentCount,
+                                                                        std::forward<FunctionT>(function)));
+    }
+
+    const RegisteredCommandBase* getRegisteredCommand(const std::string& commandName) const {
```

Comment:
If an exception is already thrown in this function, than I would return by reference and not by pointer.  If you wouldn't throw exceptions here, than a pointer could be alright because you can check if it null or not.  An optional might also be applicable but is a bit inconvenient to use, depending on if you want to return by reference or by value